src/js/webrequest.js

```diff
@@ -887,17 +905,57 @@ function dispatcher(request, sender, sendResponse) {
   }
 }
 
+
+/**
+ * Cancels all requests, populates list of pending tabs to be reloaded later,
+ * when Privacy Badger is ready.
+ *
+ * @param {Object} details chrome.webRequest request details
+ * @returns {Object}
+ */
+function onBeforeReady(details) {
+  if (!pendingTabs || details.tabId < 0) {
+    return;
+  }
+
+  pendingTabs.add(details.tabId);
+
+  return {
+    cancel: true
+  };
+}
+
+/**
+ * Reloads all tabs that loaded before Privacy Badger was ready.
+ */
+function reloadPendingTabs() {
+  for (let tab_id of pendingTabs) {
+    chrome.tabs.reload(tab_id);
+  }
+  pendingTabs = null;
+}
+
+
 /*************** Event Listeners *********************/
+function startBeforeReadyListener() {
+  chrome.webRequest.onBeforeRequest.addListener(onBeforeReady, {urls: ["<all_urls>"]}, ["blocking"]);
+}
+
+function stopBeforeReadyListener() {
+  chrome.webRequest.onBeforeRequest.removeListener(onBeforeReady);
```

Comment:
Should the listener not be removed for it to be treated as "persistent" by Firefox? See https://github.com/uBlockOrigin/uBlock-issues/issues/128#issuecomment-435594531